tests/linen/linen_attention_test.py

```diff
@@ -102,16 +102,17 @@ def test_decoding(self, spatial_shape, attn_dims):
     decode_module = module.clone(decode=True)
 
     initial_vars = decode_module.init(key2, inputs)
+    state, params = initial_vars.pop('params')
     causal_mask = nn.attention.make_causal_mask(jnp.ones((bs,) + spatial_shape))
     y_ref = jax.jit(lambda x, y: module.apply(initial_vars, x, y))(
         inputs, causal_mask)
     # feed the inputs sequentially to simulate decoding
-    def body_fn(vars_in, x):
-      y, vars_out = decode_module.apply(vars_in, x,
-                                        mutable=['cache'])
-      return vars_out, y
+    def body_fn(cache_in, x):
```

Comment:
It's a bit funny this way, because the argument isn't a cache, it's a dict with a 'cache' key. I know it adds one LOC but I would imagine that it's clearer if you make this argument actually the cache, then avoid `**cache_in` and replace `cache_out` with `vars_out` and then return `vars_out['cache']`. WDYT?

Reply:
Yeah so cache is the wrong name here. It should be called state if it has a dict with the cache. Or just be cache. Let me fix that